flink-table/flink-table-storage/src/test/java/org/apache/flink/table/storage/filestore/utils/TestRecordReader.java

```diff
@@ -133,4 +146,154 @@ public KeyValue current() {
+                    iterator.current().value().getLong(0));
+        }
+
+        public static TestKeyValue from(
+                CombinePolicy combinePolicy, RecordIterator<KeyValue> iterator) {
+            return combinePolicy == CombinePolicy.DEDUPLICATE
+                    ? new TestKeyValue(
+                            iterator.current().key().getInt(0), iterator.current().valueKind())
+                    : new TestKeyValue(
+                            iterator.current().key().getInt(0),
+                            iterator.current().valueKind(),
+                            iterator.current().value().getLong(0));
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+
+            if (!(o instanceof TestKeyValue)) {
+                return false;
+            }
+
+            TestKeyValue that = (TestKeyValue) o;
+
+            return new EqualsBuilder()
+                    .append(sequenceNumber, that.sequenceNumber)
+                    .append(key, that.key)
+                    .append(valueKind, that.valueKind)
+                    .append(value, that.value)
+                    .isEquals();
+        }
+
+        @Override
+        public int hashCode() {
+            return new HashCodeBuilder(17, 37)
+                    .append(key)
+                    .append(valueKind)
+                    .append(sequenceNumber)
+                    .append(value)
+                    .toHashCode();
+        }
+
+        @Override
+        public String toString() {
+            return "TestKeyValue{"
+                    + "key="
+                    + key
+                    + ", valueKind="
+                    + valueKind
+                    + ", sequenceNumber="
+                    + sequenceNumber
+                    + ", value="
+                    + value
+                    + '}';
+        }
+    }
+
+    public static void testCombine(
+            CombinePolicy combinePolicy,
+            List<TestKeyValue> inputs,
+            List<TestKeyValue> expected,
+            int batchSize)
+            throws IOException {
+        TestRecordReader reader = new TestRecordReader(inputs, batchSize);
+        RecordReader<KeyValue> combineReader =
+                combinePolicy.combine(
+                        reader, USER_KEY_COMPARATOR, USER_KEY_SERIALIZER, VALUE_SERIALIZER);
+        List<TestKeyValue> values = new ArrayList<>();
+        RecordIterator<KeyValue> batch;
+        while ((batch = combineReader.readBatch()) != null) {
+            while (batch.advanceNext()) {
+                values.add(TestKeyValue.from(combinePolicy, batch));
+            }
+        }
+        combineReader.close();
+        Assert.assertTrue(reader.isClosed());
+        Assert.assertEquals(expected, values);
+    }
+
+    public static void testSortMerge(List<List<TestKeyValue>> inputs, int batchSize)
```

Comment:
Can you add a `testSortMergeWithCombiner`:
`List<List<TestKeyValue>> inputs, int batchSize, CombinePolicy combinePolicy, List<TestKeyValue> expected`,